### ISSUE TITLE
[TravisCI][IntelliJ] IntelliJ IDEA IC-182.3208.16 is used for building 2018.1

### ISSUE THREAD

**Issue by wezhang** [Microsoft/azure-tools-for-java#1732]
TravisCI uses the wrong IDEA version(182.3208.16) to build 2018.1:

```
+./gradlew clean buildPlugin --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij -s -Pintellij_version=IC-2018.1 -Pdep_plugins=org.intellij.scala:2018.1.8 --info
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
...
Resource missing. [HTTP GET: https://www.jetbrains.com/~urlswitch/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/LATEST-EAP-SNAPSHOT/maven-metadata.xml]
Creating new cache for metadata-2.23/artifact-at-url, path /home/travis/.gradle/caches/modules-2/metadata-2.23/artifact-at-url.bin, access org.gradle.cache.internal.DefaultCacheAccess@17e9bc9e
Download https://www.jetbrains.com/~urlswitch/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/LATEST-EAP-SNAPSHOT/ideaIC-LATEST-EAP-SNAPSHOT.pom
Creating new cache for metadata-2.23/module-artifacts, path /home/travis/.gradle/caches/modules-2/metadata-2.23/module-artifacts.bin, access org.gradle.cache.internal.DefaultCacheAccess@17e9bc9e
Download https://www.jetbrains.com/~urlswitch/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/LATEST-EAP-SNAPSHOT/ideaIC-LATEST-EAP-SNAPSHOT.zip
IntelliJ IDEA dependency cache directory: /home/travis/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/be099a2c25b1e63cae5d05f7f53b3e4fb9a9c551/ideaIC-LATEST-EAP-SNAPSHOT
Configuring IntelliJ IDEA extra dependencies []
IntelliJ IDEA IC-182.3208.16 is used for building
Configuring IntelliJ IDEA plugin dependencies
Configuring IntelliJ tests tasks
FAILURE: Build failed with an exception.
* Where:
Build file '/home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle' line: 22
* What went wrong:
A problem occurred evaluating root project 'azure-toolkit-for-intellij'.
> org/jetbrains/kotlin/cli/common/PropertiesKt
* Try:
Run with --debug option to get more log output.
* Exception is:
org.gradle.api.GradleScriptException: A problem occurred evaluating root project 'azure-toolkit-for-intellij'.
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:92)
```

Here is the local build info:
```
IntelliJ IDEA dependency cache directory: /home/zhwe/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2018.1/d082978a55b398ef544e7daee1dd6228b88dbfc1/ideaIC-2018.1
Configuring IntelliJ IDEA extra dependencies []
IntelliJ IDEA IC-181.4203.550 is used for building
Configuring IntelliJ IDEA plugin dependencies
Configuring IntelliJ plugin maven
Looking for builtin maven in /home/zhwe/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2018.1/d082978a55b398ef544e7daee1dd6228b88dbfc1/ideaIC-2018.1
Configuring IntelliJ plugin org.intellij.scala:2018.1.8
```

IntelliJ IDEA IC-181.4203.550 is used for building

**Comment by wezhang** [comment]
Full log of Travis CI:

```
+./gradlew clean buildPlugin --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij -s -Pintellij_version=IC-2018.1 -Pdep_plugins=org.intellij.scala:2018.1.8 --info
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
Initialized native services in: /home/travis/.gradle/native
Creating new cache for fileHashes, path /home/travis/.gradle/caches/3.5/fileHashes/fileHashes.bin, access org.gradle.cache.internal.DefaultCacheAccess@32c726ee
Creating new cache for jvmRuntimeClassSignatures, path /home/travis/.gradle/caches/3.5/fileHashes/jvmRuntimeClassSignatures.bin, access org.gradle.cache.internal.DefaultCacheAccess@32c726ee
Creating new cache for plugin-use-metadata, path /home/travis/.gradle/caches/3.5/plugin-resolution/plugin-use-metadata.bin, access org.gradle.cache.internal.DefaultCacheAccess@3700ec9c
Creating new cache for client-status, path /home/travis/.gradle/caches/3.5/plugin-resolution/client-status.bin, access org.gradle.cache.internal.DefaultCacheAccess@3700ec9c
Starting Build
Settings evaluated using settings file '/home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-intellij/settings.gradle'.
Projects loaded. Root project using build file '/home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle'.
Included projects: [root project 'azure-toolkit-for-intellij']
Evaluating root project 'azure-toolkit-for-intellij' using build file '/home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle'.
Creating new cache for metadata-1.1/results, path /home/travis/.gradle/caches/transforms-1/metadata-1.1/results.bin, access org.gradle.cache.internal.DefaultCacheAccess@4afd21c6
Creating new cache for metadata-2.23/module-metadata, path /home/travis/.gradle/caches/modules-2/metadata-2.23/module-metadata.bin, access org.gradle.cache.internal.DefaultCacheAccess@17e9bc9e
Creating new cache for metadata-2.23/artifact-at-repository, path /home/travis/.gradle/caches/modules-2/metadata-2.23/artifact-at-repository.bin, access org.gradle.cache.internal.DefaultCacheAccess@17e9bc9e
Using 2 worker leases.
Creating new cache for fileHashes, path /home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-intellij/.gradle/3.5/taskHistory/fileHashes.bin, access org.gradle.cache.internal.DefaultCacheAccess@122635ef
Configuring IntelliJ IDEA gradle plugin
Configuring patch plugin.xml task
Configuring prepare IntelliJ sandbox task
Configuring prepare IntelliJ sandbox task
Configuring plugin verification task
Configuring run IntelliJ task
Configuring building IntelliJ IDEA plugin task
Configuring publishing IntelliJ IDEA plugin task
Configuring IntelliJ resources task
Configuring IntelliJ compile tasks
Configuring IntelliJ IDEA dependency
Using IDE from remote repository
Resource missing. [HTTP GET: https://www.jetbrains.com/~urlswitch/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/LATEST-EAP-SNAPSHOT/maven-metadata.xml]
Creating new cache for metadata-2.23/artifact-at-url, path /home/travis/.gradle/caches/modules-2/metadata-2.23/artifact-at-url.bin, access org.gradle.cache.internal.DefaultCacheAccess@17e9bc9e
Download https://www.jetbrains.com/~urlswitch/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/LATEST-EAP-SNAPSHOT/ideaIC-LATEST-EAP-SNAPSHOT.pom
Creating new cache for metadata-2.23/module-artifacts, path /home/travis/.gradle/caches/modules-2/metadata-2.23/module-artifacts.bin, access org.gradle.cache.internal.DefaultCacheAccess@17e9bc9e
Download https://www.jetbrains.com/~urlswitch/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/LATEST-EAP-SNAPSHOT/ideaIC-LATEST-EAP-SNAPSHOT.zip
IntelliJ IDEA dependency cache directory: /home/travis/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/be099a2c25b1e63cae5d05f7f53b3e4fb9a9c551/ideaIC-LATEST-EAP-SNAPSHOT
Configuring IntelliJ IDEA extra dependencies []
IntelliJ IDEA IC-182.3208.16 is used for building
Configuring IntelliJ IDEA plugin dependencies
Configuring IntelliJ tests tasks
FAILURE: Build failed with an exception.
* Where:
Build file '/home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle' line: 22
```

**Comment by wezhang** [comment]
Looking at gradle Intellij plugin:

```groovy
    private static void configureIntellijDependency(@NotNull Project project,
                                                    @NotNull IntelliJPluginExtension extension) {
        LOG.info("Configuring IntelliJ IDEA dependency")
        def resolver = new IdeaDependencyManager(extension.intellijRepo ?: DEFAULT_INTELLIJ_REPO)
        def ideaDependency
        if (extension.localPath != null) {
            if (extension.version != null) {
                LOG.warn("Both `localPath` and `version` specified, second would be ignored")
            }
            LOG.info("Using path to locally installed IDE: '${extension.localPath}'")
            ideaDependency = resolver.resolveLocal(project, extension.localPath, extension.localSourcesPath)
        } else {
            LOG.info("Using IDE from remote repository")
            def version = extension.version ?: DEFAULT_IDEA_VERSION
            ideaDependency = resolver.resolveRemote(project, version, extension.type, extension.downloadSources,
                    extension.extraDependencies)
        }
...
```

DEFAULT_IDEA_VERSION is `LATEST-EAP-SNAPSHOT`

**Comment by Eskibear** [comment]
A good catch. But this piece of code hasn't been changed for months. It should be some place else. E.g. `extension.version` is not specified for some reason.

**Comment by wezhang** [comment]
The `extension` is configured in the `intellij {...}` block of `build.gradle`.

**Comment by Eskibear** [comment]
> org/jetbrains/kotlin/cli/common/PropertiesKt

fixed by upgrade kotlin plugin to v1.2.50
